src/main/java/microsoft/exchange/webservices/data/core/ExchangeServiceBase.java

```diff
@@ -888,15 +893,14 @@ private void saveHttpResponseHeaders(Map<String, String> headers) {
    * @return session key
    */
   public static byte[] getSessionKey() {
+    return ExchangeServiceBase.binarySecret;
+  }
+  
+  private static byte[] generateSessionKey(){
     // this has to be computed only once.
-    synchronized (ExchangeServiceBase.class) {
-      if (ExchangeServiceBase.binarySecret == null) {
-        Random randomNumberGenerator = new Random();
-        ExchangeServiceBase.binarySecret = new byte[256 / 8];
-        randomNumberGenerator.nextBytes(binarySecret);
-      }
-
-      return ExchangeServiceBase.binarySecret;
-    }
+	Random randomNumberGenerator = new SecureRandom();
+	final byte[] bytes = new byte[256 / 8];
```

Comment:
As you noted, I only refactored this magic number containing code; I don't know the significance of the magic numbers (256 and 8). Do you know or do you know how I can find out so I make them into appropriately named constants?

I see these lines were added in https://github.com/OfficeDev/ews-java-api/blame/45e3fe6301e5824a1186e9e279200df5d57d73fe/src/main/java/microsoft/exchange/webservices/data/ExchangeServiceBase.java but the commit comment ("Initial Commit") isn't helpful, and I don't know where else to look.